jl4-decision-service/src/Backend/Jl4.hs

```diff
@@ -92,12 +93,16 @@ literalToExpr = \case
 
 valueToFnLiteral :: (Monad m) => Eval.Value -> ExceptT EvaluatorError m FnLiteral
 valueToFnLiteral = \case
-  Eval.ValNumber i -> pure $ FnLitInt $ fromIntegral i
+  Eval.ValNumber i ->
+    if denominator i == 1
+      then pure $ FnLitInt $ numerator i
+      else pure $ FnLitDouble $ fromRational i
   Eval.ValString t -> pure $ FnLitString t
   Eval.ValList vals -> do
     lits <- traverse valueToFnLiteral vals
     pure $ FnArray lits
   Eval.ValClosure {} -> throwE $ InterpreterError "#EVAL produced function closure."
+  Eval.ValUnaryBuiltinFun {} -> throwE $ InterpreterError "#EVAL produced function closure."
```

Comment:
I think it would be good if we could distinguish the error messages, so maybe 
```suggestion
  Eval.ValUnaryBuiltinFun {} -> throwE $ InterpreterError "#EVAL produced builtin function closure."
```